src/mscorlib/src/System/String.cs

```diff
@@ -293,6 +293,23 @@ unsafe internal int GetBytesFromEncoding(byte* pbNativeBuffer, int cbNativeBuffe
             }            
         }
 
+        unsafe internal int GetByteCountFromEncoding(int index, int count, Encoding encoding)
+        {
+            fixed (char* pwzChar = &this.m_firstChar)
```

Comment:
`string.m_firstChar` is private and cannot be taken. I can it move to Encoding but it will be old code with buffers and coping.